crates/hyperdrive-math/src/lp.rs

```diff
@@ -600,6 +621,540 @@ impl State {
+    ///
+    /// 1. If `y_s * t_s <= y_l * t_l` or
+    ///    `y_max_out(I) >= y_s * t_s - y_l * t_l`, set `dz_max = I` and
+    ///    proceed to step (3). Otherwise, proceed to step (2).
+    /// 2. Solve `y_max_out(dz_max) = y_s * t_s - y_l * t_l` for `dz_max`
+    ///    using Newton's method.
+    /// 3. Set `dw = (1 - PV(dz_max) / PV(0)) * l`. If `dw <= w`, then
+    ///    proceed to step (5). Otherwise, set `dw = w` and continue to
+    ///    step (4).
+    /// 4. Solve `PV(0) / l = PV(dz) / (l - dw)` for `dz` using Newton's
+    ///    method if `y_l * t_l != y_s * t_s` or directly otherwise.
+    /// 5. Return `dw` and `dz`.
+    ///
+    ///  Returns (withdrawal_shares_redeemed, share_proceeds)
+    pub fn calculate_distribute_excess_idle(
+        &self,
+        current_block_timestamp: U256,
+        active_lp_total_supply: FixedPoint,
+        withdrawal_shares_total_supply: FixedPoint, // withdraw shares - ready to withdraw
+        max_iterations: u64,
+    ) -> Result<(FixedPoint, FixedPoint)> {
+        // Steps 1 and 2: Calculate the maximum amount the share reserves can be
+        // debited. If the effective share reserves or the maximum share
+        // reserves delta can't be calculated or if the maximum share reserves
+        // delta is zero, idle can't be distributed.
+        // doesn't need original effective share reserves, gets it from state
+        let max_share_reserves_delta =
+            self.calculate_max_share_reserves_delta(current_block_timestamp)?;
+
+        if max_share_reserves_delta == fixed!(0) {
+            return Err(eyre!("Max share reserves delta is zero."));
+        }
+
+        // Step 3: Calculate the amount of withdrawal shares that can be
+        // redeemed given the maximum share reserves delta.  Otherwise, we
+        // proceed to calculating the amount of shares that should be paid out
+        // to redeem all of the withdrawal shares.
+        let withdrawal_shares_redeemed = {
+            let withdrawal_shares_redeemed = self
+                .calculate_distribute_express_idle_withdrawal_shares_redeemed(
+                    current_block_timestamp,
+                    max_share_reserves_delta,
+                    active_lp_total_supply,
+                    withdrawal_shares_total_supply,
+                )?;
+
+            // Step 3: If none of the withdrawal shares could be redeemed, then
+            // we're done and we pay out nothing.
+            if withdrawal_shares_redeemed == fixed!(0) {
+                return Err(eyre!(
+                    "Withdrawal shares redeemed from excess idle is zero."
+                ));
+
+            // Step 3: Otherwise if this amount is less than or equal to the amount
+            // of withdrawal shares outstanding, then we're done and we pay out the
+            // full maximum share reserves delta.
+            } else if withdrawal_shares_redeemed <= withdrawal_shares_total_supply {
```

Comment:
preferred formatting on is to separate the closing bracket from the "else if":

```rs
if thing {
  ...
}

// comment
else if other_thing {
  ...
}

// comment
else {
  ...
}
```

Reply:
e.g. here: https://github.com/delvtech/hyperdrive-rs/blob/main/crates/hyperdrive-math/src/long/targeted.rs#L81